packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -59,15 +59,23 @@ def self.turn_off_resource_bundle_react_core(installer)
     end
 
     def self.exclude_i386_architecture_while_using_hermes(installer)
+        isUsingHermes = self.has_pod(installer, 'hermes-engine')
+        key = "EXCLUDED_ARCHS[sdk=iphonesimulator*]"
+
         projects = self.extract_projects(installer)
 
         projects.each do |project|
             project.build_configurations.each do |config|
-                currentValue = config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] || ""
+                current_setting = config.build_settings[key] || ""
+
+                excludesI386 = current_setting.include?("i386")
 
-                unless currentValue.include?("i386")
+                if isUsingHermes and !excludesI386
                     # Hermes does not support `i386` architecture
-                    config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "#{currentValue} i386".strip
+                    config.build_settings[key] = "#{current_setting} i386".strip
+                else
+                    # leave EXCLUDED_ARCHS unchanged or set it to "" if it doesn't already exist
+                    config.build_settings[key] = current_setting
```

Comment:
Can we remove this `else`? I think it's better not to touch the settings to avoid introducing unexpected side effects, if we don't want to change them.
Also, less code, means less maintenance required and a lower bug probability.

Reply:
done ✅